velocity_modelling/scripts/plot_velocity_model.py

```diff
@@ -0,0 +1,224 @@
+from enum import Enum
+from pathlib import Path
+from typing import Annotated, Optional
+
+import numpy as np
+import pandas as pd
+import typer
+import yaml
+
+from pygmt_helper import plotting
+from qcore import grid
+from velocity_modelling import bounding_box
+
+app = typer.Typer()
+
+
+class VelocityModelComponent(str, Enum):
+    p_wave = "p_wave"
+    s_wave = "s_wave"
+    density = "rho"
+
+
+@app.command()
+def plot_velocity_model(
+    output_plot_path: Annotated[
+        Path,
+        typer.Argument(
+            help="Path to output velocity model plot.", writable=True, dir_okay=False
+        ),
+    ],
+    vm_params_ffp: Annotated[
+        Optional[Path], typer.Option(help="Path to VM params YAML", dir_okay=False)
+    ] = None,
+    velocity_model_ffp: Annotated[
```

Comment:
Surely this path shouldnt be optional, otherwise your just plotting a blank map and this script is specific to plotting the velocity model

Reply:
It's optional because sometimes you want to plot the boundary of the domain without the velocity model over-top. I do this quite frequently to check that the domain makes sense. Cesar also does this when checking that the fault he has is contained inside the domain.

Reply:
Specifically in Cesar's case, he often *can't* supply a velocity model because the VM gen step broke because the domain is busted. The primary motivation for the script is to generate plots for that use-case. I added plotting of the actual velocity model on-top as a useful check that the velocity model makes sense. Plotting the velocity model values has caught some bugs in my velocity model generation as well, so both make sense to me.